### DISCUSSION TITLE
「コンポーザブルとは何か ?」の質問

### DISCUSSION THREAD

**Discussion by tokuda109** [vuejs-jp/learn.nuxt.com#73] [ハンズオン Q&A] [answered]
https://github.com/vuejs-jp/learn.nuxt.com/blob/main/content/1.vue/4.composition-api/index.md で何か困ったことや質問があれば遠慮なく質問して下さい

**Comment by nanauda** [comment]
app.vue内でuseCounter()は、import無しで使用できるのでしょうか?(これはNuxtの機能の一部ですか?)
その場合、composable配下に置くことで、それらの配下のすべてのexportされた関数が、import無しで使用できると理解していいですか?

**Reply by atlansien** [reply]
@nanauda 
ご質問ありがとうございます！はい、認識の通り、composable配下でexportされた関数は  .vue 内でimport無しで利用することができます。

**Reply by atlansien** [reply]
こちらの件についてはこの後のセッションでも解説がありますのでぜひお楽しみにしてくださいmm

**Reply by nanauda** [reply]
承知しました！ご回答ありがとうございます。

**Comment by nbifrye** [comment]
nuxt を使う上で、ユーティリティ `utils/*` とコンポーザブル `composables/*` について、「Vue の API を扱うか否か」や「状態を持つか否か」など、使い分けのプラクティスあれば知りたいです

**Reply by nbifrye** [reply, marked as answer]
回答いただきありがとうございました！

- Vue の API (例: Composition API, `ref()` など) を使うようなものをまとめたロジック群は `composables/*` に
- Vue に閉じないようなロジックは `utils/*` に
  - 例: スネークケース/キャメルケースの変換など
- Nuxt 自身の API を参考にするとよい
  - https://nuxt.com/docs/api
